Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -417,37 +417,37 @@ private HttpClient CreateClient()
         return client;
     }
 
-    public async Task Push(Image x, string name, string? tag, string baseName, Action<string> logProgressMessage)
+    public async Task Push(Image x, ImageReference source, ImageReference destination, Action<string> logProgressMessage)
     {
-        tag ??= "latest";
 
         HttpClient client = GetClient();
-        var reg = this;
+        Registry destinationRegistry = (destination.Registry!).Value;
```

Comment:
we only ever push to real registries - local daemons go through a different code path, so this force unwrapping is safe.